tiledb-sys/src/datatype.rs

```diff
@@ -224,10 +316,80 @@ impl Display for Datatype {
 
 #[cfg(test)]
 mod tests {
+    use super::*;
+
     #[test]
     fn datatype_test() {
         for i in 0..256 {
-            println!("{}", i);
+            println!("I: {}", i);
+            if i <= 43 {
+                let dt = Datatype::from_u32(i as u32)
+                    .expect("Error converting value to Datatype");
+                assert_ne!(
+                    format!("{}", dt),
+                    "<UNKNOWN DATA TYPE>".to_string()
+                );
+                assert!(check_valid(&dt));
```

Comment:
I like the assertion that only one physical type is valid for each logical type but the way this is written leaves a lot of room for failure if the wrong one is chosen.  I think it's probably worth the map from logical to physical type (which we probably want somewhere anyway?) and then having the `check_valid` instead return a vector representing which types were valid.

Reply:
For sure. The only guarantee that we've picked the right type is to have the second map but I didn't have the brain power to figure out how best to do that in Rust. I'm more than open to adding it if you've got any idea on how to write it.

Reply:
I'm happy to check out this branch and give it a whirl, or do so after you've merged